rtengine/rt_algo.cc

```diff
@@ -276,17 +288,72 @@ void buildBlendMask(float** luminance, float **blend, int W, int H, float &contr
+                        // Means we scan (2 * skip + 1)^2 tiles in this step to get a better hit rate
+                        // fortunately the scan is quite fast, so we use only one core and don't parallelize
+                        const int topLeftYStart = std::max(minY - skip, 0);
+                        const int topLeftXStart = std::max(minX - skip, 0);
+                        const int topLeftYEnd = std::min(minY + skip, H - tilesize);
+                        const int topLeftXEnd = std::min(minX + skip, W - tilesize);
+                        const int numTilesH = topLeftYEnd - topLeftYStart + 1;
+                        const int numTilesW = topLeftXEnd - topLeftXStart + 1;
+
+                        std::vector<std::vector<float>> variances(numTilesH, std::vector<float>(numTilesW));
+                        for (int i = 0; i < numTilesH; ++i) {
+                            const int tileY = topLeftYStart + i;
+                            for (int j = 0; j < numTilesW; ++j) {
+                                const int tileX = topLeftXStart + j;
+                                const float avg = tileAverage(luminance, tileY, tileX, tilesize);
+
+                                if (avg < minLuminance || avg > maxLuminance) {
+                                    // too dark or too bright => skip the tile
+                                    variances[i][j] = RT_INFINITY_F;
+                                    continue;
+                                } else {
+                                    variances[i][j] = tileVariance(luminance, tileY, tileX, tilesize, avg);
+                                }
+                            }
+                        }
+
+                        float minvar = RT_INFINITY_F;
+                        int minI = 0, minJ = 0;
+                        for (int i = 0; i < numTilesH; ++i) {
+                            for (int j = 0; j < numTilesW; ++j) {
+                                if (variances[i][j] < minvar) {
+                                    minvar = variances[i][j];
+                                    minI = i;
+                                    minJ = j;
+                                }
+                            }
+                        }
+                        if (minvar <= 4.f) {
+                            JaggedArray<float> Lum(tilesize, tilesize);
+                            JaggedArray<float> Blend(tilesize, tilesize);
+                            const int minY = topLeftYStart + minI;
+                            const int minX = topLeftXStart + minJ;
+                            for (int i = 0; i < tilesize; ++i) {
+                                for (int j = 0; j < tilesize; ++j) {
+                                    Lum[i][j] = luminance[i + minY][j + minX];
+                                }
+                            }
+                            contrastThreshold = calcContrastThreshold(Lum, Blend, tilesize, tilesize) / 100.f;
```

Comment:
Isn't that a repetition of the above? I mean: Shouldn't that go into a function or lambda?